src/utils/walk.ts

```diff
@@ -33,41 +33,39 @@ function assignNodeSpecificFields(base: IBaseNode, node: JSONSchema4) {
   }
 }
 
-function getType(node: JSONSchema4) {
-  if (Array.isArray(node.type)) {
-    return node.type.length === 1 ? node.type[0] : node.type;
+function inferType(node: JSONSchema4): JSONSchema4TypeName | undefined {
+  if ('properties' in node) {
+    return SchemaKind.Object;
   }
 
-  return node.type;
-}
+  if ('items' in node) {
+    return SchemaKind.Array;
+  }
 
-function processNode(node: JSONSchema4): SchemaNode | void {
-  const combiner = getCombiner(node);
+  return;
+}
 
-  if (node.type !== undefined && combiner === undefined) {
-    const base: IBaseNode = {
-      id: assignId(node),
-      type: getType(node),
-      validations: getValidations(node),
-      annotations: getAnnotations(node),
-      enum: node.enum,
-    };
+function getPrimaryType(node: JSONSchema4) {
```

Comment:
what if type is `["object", "array"]`

Reply:
Really the user should be using an oneOf like below, but if having both in type is valid we definitely need to support it

```
{
    "oneOf": [
        {"$ref": "#/definitions/singleObject"}, // plain object
        {
            "type": "array", // array of plain objects
            "items": {"$ref": "#/definitions/singleObject"}
        }
    ],
    "definitions": {
        "singleObject": {...}
    }
}```

Reply:
This also means that if type is `["object", "array"]` we need to infer that its a combiner type oneOf so it renders properly in the ui

Reply:
Yeah, shared the same concern here https://github.com/stoplightio/json-schema-viewer/pull/17#discussion_r283318479.
I'm not sure if displaying `oneOf` is safe though. To me, it sounds like a nice idea, but we'll need to make it reliable (transform the data correctly).
 @marbemac, what's your take on this?

Reply:
I think we don't worry about this use case until a customer complains about it.